src/plugins/recursion/README.md

```diff
@@ -0,0 +1,64 @@
+
+
+## Introduction
+
+I took [Relax NG](http://relaxng.org/spec-20011203.html#define-ref)'s `ref` and `define` part and tried to "copy" 
+their approach for elektra's specification.
+Until now I haven't really gotten a sensible idea for that though. Maybe we could talk on the next meeting about this.
+
+Also I have no idea what your `struct` plugin actually does. There is no example, just some code snippet from 
+the contract which doesn't help understanding the plugin.
+
+## Problems
+1) We want to check if all recursions are non-cyclical
+2) We want to check if a recursive config actually leads to another correct config and not into NULL for example
+3) Forcing users to define non-recursive settings first on which they can build up 
+recursive settings is inconvenient and could lead to orphan settings.
+4) We can only check right after calling `kdb set` and not a whole config. 
+This means that user can have "unfinished" configurations while setting new values.
+
+## Possible Solution
+
+Users have to define recursive setting specifications in the following way:
+
+```
+kdb mount config.dump /recursive dump recursion
+kdb setmeta user/recursive check/recursion/vertex "menu, submenu"
+kdb setmeta user/recursive check/recursion/leaf "param, menupoint"
```

Comment:
Please use Elektra arrays, for example:

```
kdb setmeta user/recursive check/recursion/vertex/#0 menu
kdb setmeta user/recursive check/recursion/vertex/#1 submenu
kdb setmeta user/recursive check/recursion/leaf/#0 param
kdb setmeta user/recursive check/recursion/leaf/#1 menupoint
```

See #1010

A `kdb set-meta-array` would be nice to have.